multicollector.go

```diff
@@ -19,42 +20,88 @@ type multiCollector struct {
 	// Impose a timeout on collection if non-zero.
 	timeout time.Duration
 
+	logger *log.Logger
+
+	warningsDesc *prometheus.Desc
+
 	members []multiCollectorMember
 }
 
 var _ prometheus.Collector = (*multiCollector)(nil)
 
 func newMultiCollector(m ...multiCollectorMember) *multiCollector {
 	return &multiCollector{
+		logger: log.Default(),
+		warningsDesc: prometheus.NewDesc("paperless_warnings_total",
+			"Number of warnings generated while scraping metrics.",
+			nil, nil),
 		members: m,
 	}
 }
 
 func (c *multiCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.warningsDesc
+
 	for _, i := range c.members {
 		i.describe(ch)
 	}
 }
 
-func (c *multiCollector) Collect(ch chan<- prometheus.Metric) {
-	ctx := context.Background()
+func (c *multiCollector) collect(ctx context.Context, ch chan<- prometheus.Metric) error {
```

Comment:
While reading the review, i was confused that we have two functions, one `collect` and one `Collect`.
Maybe renaming this one to another name (I don't have an alternative proposal right now).